File: backend/internal/domain/category.go
package domain

import "errors"

// ErrDuplicateCategoryName は同一ユーザー内でカテゴリ名が重複している場合のエラーです。
var ErrDuplicateCategoryName = errors.New("category name already exists")

// ErrCategoryNotFound はカテゴリが存在しない場合のエラーです。
var ErrCategoryNotFound = errors.New("category not found")

type Category struct {
	ID           int    `json:"id"`
	Name         string `json:"name"`
	CategoryType string `json:"category_type"`
	SortOrder    int    `json:"sort_order"`
}

File: frontend/src/lib/types/category.ts
export type Category = {
  id: number
  name: string
  category_type: 'default' | 'user' | 'other'
  sort_order: number
}

export type CreateCategoryInput = {
  name: string
}

export type UpdateCategoryInput = {
  name: string
}

export type ReorderCategoryItem = {
  id: number
  sort_order: number
}